stan/math/opencl/rev/arena_matrix_cl.hpp

```diff
@@ -39,10 +60,88 @@ class arena_matrix_cl : public chainable_alloc, public matrix_cl<T> {
    * @tparam Expr expression type
    * @param expression expression
    */
+  // we need this as a separate overload, because the general constructor is
+  // explicit
```

Comment:
I would just include this in the regular doc

Reply:
This is just an implementation detail. It is not useful for whoever is using `arena_matrix_cl`.